rubicon/objc/types.py

```diff
@@ -18,17 +18,194 @@
 __arm64__ = (_any_arm and __LP64__)
 __arm__ = (_any_arm and not __LP64__)
 
-PyObjectEncoding = b'{PyObject=@}'
 
-
-def encoding_for_ctype(vartype):
-    typecodes = {
-        c_char: b'c', c_int: b'i', c_short: b's', c_long: b'l', c_longlong: b'q',
-        c_ubyte: b'C', c_uint: b'I', c_ushort: b'S', c_ulong: b'L', c_ulonglong: b'Q',
-        c_float: b'f', c_double: b'd', c_bool: b'B', c_char_p: b'*', c_void_p: b'@',
-        py_object: PyObjectEncoding
-    }
-    return typecodes.get(vartype, b'?')
+_type_to_ctype_map = {
+    int: c_int,
+    float: c_float,
+    bool: c_bool,
+    # This is not really correct - Python strings are Unicode - but we need to keep this for compatibility.
```

Comment:
Yes, Python's int is arbitrary precision, but there's an issue of practicality; int is still a common datatype. Providing *a* mapping for int, even if it's one with known limitations, would seem preferable to not providing a mapping at all.

As for float - modifying it to point to c_double would seem to make sense to me, rather than removing it entirely.

Neither change would affect the `NSInteger` et al use case, but it would preserve basic functionality for a non-zero number of cases.